src/helpers/ConnectorConfigInput.tsx

```diff
@@ -288,7 +288,7 @@ class ConnectorConfigInput {
           // Check if the child has another level or not
           if (ele.specs && ele.enum && ele.enum.length === 0) {
             ele.specs.forEach((childEle: ConnectorSpec) => {
-              filteredChildSpecs.push({ ...childEle, order: ele.order });
+              filteredChildSpecs.push({ ...childEle, parent: childEle?.parent && childEle?.parent?.length > 0 ? childEle.parent : ele.parent, order: ele.order });
             });
```

Comment:
Hey @huzaifmalik786 why is this needed here `&& childEle?.parent?.length` . Wont the condition before `&&` handle all cases of `null`, `undefined` and empty string. 

Rest looks good to me !!

Reply:
@Ishankoradia , i saw that the child element has empty string as parent. So, it's there to handle the case of empty string i.e. "".

Reply:
My bad. It is working. I'll change it.